plugin/core/configurations.py

```diff
@@ -86,32 +86,22 @@ def _create_window_configs(self) -> List[ClientConfig]:
     def _apply_project_overrides(self, client_config: ClientConfig, project_clients: Dict[str, Any]) -> ClientConfig:
         overrides = project_clients.get(client_config.name)
         if overrides:
-            debug('window has override for {}'.format(client_config.name), overrides)
-            client_settings = _merge_dicts(client_config.settings, overrides.get("settings", {}))
-            client_env = _merge_dicts(client_config.env, overrides.get("env", {}))
+            debug('applying .sublime-project override for', client_config.name)
+            client_config.settings.update(overrides.get("settings", {}))
+            for key, value in overrides.get("env", {}).items():
+                client_config.env[key] = value
```

Comment:
Why not using `update` also for `env`?

Reply:
Because `env` is a `dict`, whereas `settings` is a `DottedDict`.